pilot/cmd/pilot-agent/main.go

```diff
@@ -54,6 +55,10 @@ import (
 	"istio.io/istio/pkg/util/gogoprotomarshal"
 )
 
+// TODO: Move most of this to pkg.
+
+// Istio-agent requires recent K8S with JWT support. This is used to authenticate with control plane
```

Comment:
You can probably call out the supported version. K8s supports trustworthy jwt from 1.12 (but 1.12 didn't support audiences different than api server audiences - this came in 1.13)

Reply:
My understanding is that the apiserver verification didn't support the audience - but issuing and checking using standard JWT library would still work, they are standard RS256 JWT tokens, singed with the K8S CA. 

I'm planning to add some code to support checking the tokens without a call to apiserver, so istiod can run outside of the cluster, I suspect this might take care of 1.12 as well.